iroha_actor/src/lib.rs

```diff
@@ -57,27 +58,37 @@ impl<A: Actor> Addr<A> {
         Self {
             sender,
             #[cfg(feature = "deadlock_detection")]
-            actor_id: ActorId::from(&task::current()),
+            actor_id: ActorId::new(Some(type_name::<Self>())),
         }
     }
 
     /// Send a message and wait for an answer.
     /// # Errors
     /// Fails if noone will send message
-    pub async fn send<M>(&self, message: M) -> Result<M::Result, RecvError>
+    #[allow(unused_variables, clippy::expect_used)]
+    pub async fn send<M, B: Actor>(
+        &self,
+        message: M,
+        from: Option<Addr<B>>,
```

Comment:
Changing api for this case is bad. Can we think about better workaround which would allow us not to have this argument?

Reply:
It is not possible in the way it was before - even if we have task local storage, is is not `actor` local in the sense that many tasks can be spawned inside of the actor.

We can add a `send_from` method as an alternative and `send` will just call it with `None` option.

Reply:
Actually we can have a task local storage... Before hand it worked like this

Reply:
Yes, I know that it worked like that, but is is semantically incorrect behavior. We should not think that actor would use only one task.

Reply:
Actually it is correct. I mean we couldn't check the case like this
```rust
struct Actor1;
struct Actor2;
impl Handler<()> for Actor1 {
  async fn handle(&mut self, (): (), _: &mut Context) {
    task::spawn(async {
       actor2.send(()).await;
    }).await;
  }
}
```

But it couldn't be checked even before. We should just forget about this weird cases as we couldn't check them even theoretically (without changing api, which is crucial). This opt-in feature should just change api.

Reply:
I mean why is not changing API crucial? I agree on the other points though. But still I suggest to do it in a separate PR. Tokio migration is already too big. Let's create issues for this with high priority.

Reply:
Though I am still a bit reluctant to support the library - deadlock detection - that we will not use at all.

Reply:
We actually will use it sometimes to request things and wait for answer from another actor, so it would make sense (take a look at the actor migration pr).

Reply:
https://jira.hyperledger.org/browse/IR-1109